src/Models/ShallowWaterModels/shallow_water_model.jl

```diff
@@ -212,21 +213,25 @@ validate_momentum_advection(momentum_advection, formulation) = momentum_advectio
 validate_momentum_advection(momentum_advection, ::VectorInvariantFormulation) =
     throw(ArgumentError("VectorInvariantFormulation requires a vector invariant momentum advection scheme. \n"* 
                         "Use `momentum_advection = VectorInvariant()`."))
-validate_momentum_advection(momentum_advection::VectorInvariantSchemes, ::VectorInvariantFormulation) = momentum_advection
+validate_momentum_advection(momentum_advection::Union{VectorInvariantSchemes, Nothing}, ::VectorInvariantFormulation) = momentum_advection
 
 formulation(model::ShallowWaterModel)  = model.formulation
 architecture(model::ShallowWaterModel) = model.architecture
 
-function shallow_water_velocities(model::ShallowWaterModel)
-    if formulation(model) isa VectorInvariantFormulation 
-        return (model.solution.u, model.solution.v) 
+# The w velocity is needed to use generic TurbulenceClosures methods, therefore it is set to nothing
+function shallow_water_velocities(solution, formulation)
+    if formulation isa VectorInvariantFormulation 
+        return (u = solution.u, v = solution.v, w = nothing) 
     else
-        u = Field(@at (Face, Center, Center) model.solution.uh / model.solution.h)
-        v = Field(@at (Center, Face, Center) model.solution.vh / model.solution.h)
+        u = Field(@at (Face, Center, Center) solution.uh / solution.h)
+        v = Field(@at (Center, Face, Center) solution.vh / solution.h)
 
         compute!(u)
         compute!(v)
 
-        return (u, v)
+        return (; u, v, w = nothing)
     end
-end
\ No newline at end of file
+end
+
+shallow_water_fields(velocities, solution, tracers, ::ConservativeFormulation)    = merge(velocities, solution, tracers)
+shallow_water_fields(velocities, solution, tracers, ::VectorInvariantFormulation) = merge(solution, (; w = velocities.w), tracers)
```

Comment:
I have `w = nothing` because all the other models have the following structure 
```
model.velocities
NamedTuple with 3 Fields on 6×6×6 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 3×3×3 halo:
├── u: 6×6×6 Field{Face, Center, Center} on RectilinearGrid on CPU
├── v: 6×6×6 Field{Center, Face, Center} on RectilinearGrid on CPU
└── w: 6×6×7 Field{Center, Center, Face} on RectilinearGrid on CPU
```
To make all methods defined in turbulence closures work for the `ShallowWaterModel` we need a `w` velocity (even if that is `nothing`). That is why that `w` is there

Reply:
Thanks for explaining.